### PR TITLE
Annotation with or without cache

### PR DESCRIPTION
**Pull request by MaxUlysse**
- add `--annotation_cache` parameter to use the annotation cache, which need to be used with:
    - `---snpEff_cache` to specify snpEff cache
    - `--vep_cache` to specify VEP cache
- VEP cache version is now specified within genome config files, and can be modified via command line if needed
- Documentation has been made about annotation
- snpEff and VEP containers are now made with conda
- removed snpEff base container
- updated snpEff and VEP
- snpEff and VEP are now also in the base container
- update docs

## PR checklist
 - [X] PR is made against `dev` branch
 - [X] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [X] Ensure the test suite passes (`./scripts/test.sh -p docker -t ALL`).
 - [X] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated
 - [X] `README.md` is updated

EDIT: Forgot the details...

### CODE REVIEW

File: docs/ANNOTATION.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Sarek has already designed containers with `snpEff` and `VEP` files for `GRCh37` and `GRCh38`.
+Default settings will run using these containers.
+
+## Using cache
+
+Both `snpEff` and `VEP` enable usage of cache.
+If cache is available on the machine where Sarek is run, it is possible to run annotation using cache.
+You can specify the cache directory using `--snpEff_cache` and `--vep_cache` in the command lines or within configuration files.
+But the cache will only be used whenn `--annotation_cache` is specified (either in command lines or in a configuration file).
```
Comment (alneberg) [inline docs/ANNOTATION.md]: Typo! ;)

File: docs/ANNOTATION.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Example:
+```
+nextflow run annotate.nf --tools snpEff --annotateVCF file.vcf.gz --genome GRCh38 --snpEff_cache /Path/To/snpEffCache --annotation_cache
+nextflow run annotate.nf --tools VEP --annotateVCF file.vcf.gz --genome GRCh38 --vep_cache /Path/To/vepCache --annotation_cache
+```
+
+## Downloading cache
+
+An helper script has been designed to help downloading snpEff and VEP cache.
```
Comment (alneberg) [inline docs/ANNOTATION.md]: You could add here as well that this is only necessary if one is not running on uppmax and do not wish to use the specific containers.

File: docs/CONTAINERS.md [hunk 3]
```diff
@@ -2,8 +2,9 @@
 
 Subsets of all containers can be downloaded:
 
-- For processing, germline and somatic variant calling and Reports:
+- For processing, germline variant calling and Reports:
```
Comment (alneberg) [inline docs/CONTAINERS.md]: It's not used for somatic VC?

Comment (MaxUlysse) [reply docs/CONTAINERS.md]: yes, that's why I said **add** r-base and runallelecount for somatic VC

File: docs/CONTAINERS.md [hunk 4]
```diff
@@ -12,6 +13,8 @@ Subsets of all containers can be downloaded:
 - For annotation for GRCh38, you will need:
   - [snpeffgrch38](#snpeffgrch38-)
   - [vepgrch38](#vepgrch38-)
+- For annotation with cache use:
+  - [sarek](#sarek-)
```
Comment (MaxUlysse) [inline docs/CONTAINERS.md]: But in that case, I could probably say, you should already have this one...

File: docs/PARAMETERS.md [hunk 5]
```diff
@@ -112,6 +111,23 @@ Possible values are:
 Choose vcf to annotate.
 Different vcfs to be separated by commas.
 
+### --annotate_cache
+
```
Comment (alneberg) [inline docs/PARAMETERS.md]: Could you clarify the edge cases, for example: using: `--annotate_cache` but not either `--snpEff_cache` or `--vep_cache`; using `--snpEff_cache` without `--annotate_cache`?

Comment (MaxUlysse) [reply docs/PARAMETERS.md]: won't work.
You need to use `--annotate_cache` with `--snpEff_cache` or/and `--vep_cache`
If you use only `--annotate_cache` it's not using cache
If you use only `--snpEff_cache` or/and `--vep_cache` it's not using cache either

File: docs/CONTAINERS.md (hunk 3 above)
Comment (alneberg) [reply docs/CONTAINERS.md]: True, I missed that. But for example FreeBayes still use the sarek container right?

Comment (alneberg) [reply docs/CONTAINERS.md]: Ah, ok I see now! Never mind this.

File: docs/PARAMETERS.md (hunk 5 above)
Comment (alneberg) [reply docs/PARAMETERS.md]: Yes, can you have that in the docs? It will not crash though, it will just not do what one thinks it does. Right?

Comment (MaxUlysse) [reply docs/PARAMETERS.md]: OK, I will specify all that then

**Comment by MaxUlysse** [comment]
Currently seems to be working well on `munin`.
I just now need to clean up the definition of params not to impact other profiles and I'll be OK to merge it too.

**Comment by MaxUlysse** [comment]
also closes #554

**Pull request merged**